app/org/sagebionetworks/bridge/models/schedules/ScheduleContext.java

```diff
@@ -0,0 +1,148 @@
+package org.sagebionetworks.bridge.models.schedules;
+
+import java.util.Map;
+import java.util.Objects;
+
+import org.joda.time.DateTime;
+import org.joda.time.DateTimeZone;
+import org.sagebionetworks.bridge.models.studies.StudyIdentifier;
+
+import com.google.common.collect.ImmutableMap;
+
+/**
+ * All the information necessary to convert a schedule into a set of tasks, on a given request. 
+ * Because some of these values derive from the user, there is a validator that is run on this object 
+ * that verifies the four required values (studyId, zone, endsOn and healthCode) are present.
+ * 
+ * @see org.sagebionetworks.bridge.validators.ScheduleContextValidator
+ */
+public final class ScheduleContext {
+    
+    private final StudyIdentifier studyId;
+    private final DateTimeZone zone;
+    private final DateTime endsOn;
+    private final Map<String,DateTime> events;
+    private final String schedulePlanGuid;
+    private final String healthCode;
+    
+    public ScheduleContext(StudyIdentifier studyId, DateTimeZone zone, DateTime endsOn, String healthCode, Map<String,DateTime> events, String schedulePlanGuid) {
+        this.studyId = studyId;
+        this.zone = zone;
+        this.endsOn = endsOn;
+        this.healthCode = healthCode;
+        this.events = (events == null) ? null : ImmutableMap.copyOf(events);
+        this.schedulePlanGuid = schedulePlanGuid; 
+    }
```

Comment:
Strongly recommend the builder pattern for this. That way, your large constructors become more readable and easier to refactor, and you can create "copy constructors" without having to create a bunch of intermediate objects.